packages/xstate-react/src/createActorContext.ts

```diff
@@ -26,46 +24,25 @@ export function createActorContext<TMachine extends AnyStateMachine>(
     compare?: (a: T, b: T) => boolean
   ) => T;
   useActorRef: () => ActorRefFrom<TMachine>;
-  Provider: (
-    props: {
-      children: React.ReactNode;
-      machine?: TMachine | (() => TMachine);
-    } & (AreAllImplementationsAssumedToBeProvided<
-      TMachine['__TResolvedTypesMeta']
-    > extends false
-      ? {
-          options: InternalMachineImplementations<
```

Comment:
`<Provider machine={someMachine.provide()} />` is workable but i think that it makes it a little bit awkward because we are repeating `someMachine` here - we already given it to create the `Provider` , it's available internally as a closure, types are bound to it === it looks like a chore that we still require it to be used directly here. 

The best idea I have so far to solve this is for the behavior to somehow declare it's available options. This way the interpreter-level stuff could accept those options in a generic way.

Reply:
Smth like this could work
```ts
<Context.Provider machine={machine => machine.provide({})}>{null}</Context.Provider>
```

That's quite a bit of `machine` in here though 🤔